src/Search/Kernel.php

```diff
@@ -207,12 +207,36 @@ public function dependencies(Application $app)
             return new Monitoring();
         };
 
-        $app['limit'] = function (Application $app) {
+        /**
+         * @internal
+         */
+        $app['limit._memory'] = function(Application $app) {
+            return MemoryLimit::mb($app['config']['process_memory_limit']);
+        });
+        /**
+         * @internal
+         */
+        $app['limit._signals'] = function(Application $app) {
+            return SignalsLimit::stopOn(['SIGINT', 'SIGTERM', 'SIGHUP']);
+        };
+
+        $app['limit.long_running'] = function (Application $app) {
             return new LoggingMiddleware(
                 new CompositeLimit(
-                    MemoryLimit::mb($app['config']['process_memory_limit']),
-                    SignalsLimit::stopOn(['SIGINT', 'SIGTERM', 'SIGHUP'])
-                ), $app['logger']);
+                    $app['limit._memory'],
+                    $app['limit._signals']
+                ),
+                $app['logger']
+            );
+        };
+
+        $app['limit.interactive'] = function (Application $app) {
+            return new LoggingMiddleware(
+                new CompositeLimit(
+                    $app['limit._signals']
+                ),
```

Comment:
Logging Middleware will work with a single limit, but if we add more limits it might be useful to keep this composite for clarity, they only get called on cli commands so I doubt the overhead is a concern